# Relatório Operacional de Pré-Decolagem - Aurora Siger

## Simulação de Telemetria e Análise Energética

## 1. Objetivo da simulação

Este notebook apresenta uma simulação computacional do processo de verificação pré-decolagem da espaçonave Aurora Siger. A partir de dados simulados de telemetria, são avaliadas as condições de segurança dos principais parâmetros monitorados, permitindo determinar se a espaçonave está pronta para decolar ou se a decolagem deve ser abortada.

Também é realizada uma análise energética, considerando a capacidade total de energia, o nível atual de carga, as perdas energéticas e o consumo elétrico estimado dos sistemas embarcados durante a fase de decolagem.

## 2. Premissas e critérios de segurança

O material fornecido apresenta os parâmetros de telemetria que devem ser analisados, porém não especifica todas as unidades de medida, valores de referência, faixas de segurança ou estados dos módulos críticos. Dessa forma, quando necessário para a construção da simulação computacional, foram adotadas premissas para estabelecer as condições consideradas seguras pelo algoritmo.

Essas premissas têm como finalidade permitir a implementação e demonstração do processo de verificação e não representam necessariamente especificações reais ou oficiais da missão Aurora Siger ou de uma espaçonave real.

| Parâmetro | Condição considerada segura |
|---|---|
| Temperatura interna | Entre 0 °C e 40 °C |
| Temperatura externa | Entre -20 °C e 80 °C |
| Integridade estrutural | 1 = estrutura íntegra |
| Nível de energia | Igual ou superior a 80% |
| Pressão dos tanques | Entre 1800 kPa e 2500 kPa |
| Status dos módulos críticos | Nominal |

## 3. Dados simulados de telemetria

Como o material da atividade não fornece valores específicos de telemetria, foram definidos valores simulados para representar uma leitura dos sensores da Aurora Siger e permitir a execução das verificações do algoritmo.

In [27]:
temperatura_interna = 24
temperatura_externa = 35
integridade_estrutural = 1
nivel_energia = 92
pressao_tanque = 2100
status_modulos_criticos = "nominal"

## 4. Verificação dos parâmetros

Cada parâmetro de telemetria é comparado com os critérios de segurança definidos para a simulação. O resultado de cada verificação é armazenado como um valor booleano (`True` ou `False`), permitindo posteriormente determinar a condição geral da espaçonave.

In [28]:
if temperatura_interna >= 0 and temperatura_interna <= 40:
    temperatura_interna_segura = True
else:
    temperatura_interna_segura = False

if temperatura_externa >= -20 and temperatura_externa <= 80:
    temperatura_externa_segura = True
else:
    temperatura_externa_segura = False

if integridade_estrutural == 1:
    integridade_estrutural_segura = True
else:
    integridade_estrutural_segura = False

if nivel_energia >= 80:
    nivel_energia_seguro = True
else:
    nivel_energia_seguro = False

if pressao_tanque >= 1800 and pressao_tanque <= 2500:
    pressao_tanque_segura = True
else:
    pressao_tanque_segura = False

if status_modulos_criticos == "nominal":
    modulos_criticos_seguros = True
else:
    modulos_criticos_seguros = False

## 4. Verificação dos parâmetros

Cada parâmetro de telemetria é comparado com os critérios de segurança definidos para a simulação. O resultado de cada verificação é armazenado como um valor booleano (`True` ou `False`), permitindo posteriormente determinar a condição geral da espaçonave.

In [29]:
if temperatura_interna_segura and temperatura_externa_segura and integridade_estrutural_segura and nivel_energia_seguro and pressao_tanque_segura and modulos_criticos_seguros:

    resultado_final = "PRONTO PARA DECOLAR"
else:
    resultado_final = "DECOLAGEM ABORTADA"

## 5. Resultado da verificação

A decolagem somente será considerada segura caso todos os parâmetros analisados atendam simultaneamente aos critérios estabelecidos. Caso pelo menos um parâmetro apresente uma condição insegura, o resultado será **"DECOLAGEM ABORTADA"**.

In [30]:
print("Temperatura interna:", temperatura_interna_segura)
print("Temperatura externa:", temperatura_externa_segura)
print("Integridade estrutural:", integridade_estrutural_segura)
print("Nível de energia:", nivel_energia_seguro)
print("Pressão dos tanques:", pressao_tanque_segura)
print("Módulos críticos:", modulos_criticos_seguros)

print("Resultado final:", resultado_final)

Temperatura interna: True
Temperatura externa: True
Integridade estrutural: True
Nível de energia: True
Pressão dos tanques: True
Módulos críticos: True
Resultado final: PRONTO PARA DECOLAR


## 6. Análise energética

Para a simulação, foram considerados uma capacidade total de 1500 kWh, carga atual de 90%, consumo elétrico estimado de 300 kWh durante a fase de decolagem e perdas energéticas de 5%.

O consumo estimado na decolagem representa a energia elétrica utilizada pelos sistemas embarcados durante essa etapa da missão, não correspondendo à energia empregada diretamente na propulsão da espaçonave.

In [31]:
capacidade_total = 1500      # kWh
carga_atual = 90             # %
consumo_decolagem = 300      # kWh
perdas_energeticas = 5       # %

energia_disponivel = (capacidade_total * carga_atual) / 100
perdas_energeticas_kWh = (energia_disponivel * perdas_energeticas) / 100
energia_util = energia_disponivel - perdas_energeticas_kWh
energia_restante = energia_util - consumo_decolagem

print(f"Energia Disponível: {energia_disponivel} kWh")
print(f"Perdas Energéticas (%): {perdas_energeticas} %")
print(f"Perdas Energéticas (kWh): {perdas_energeticas_kWh} kWh")
print(f"Energia Útil: {energia_util} kWh")
print(f"Energia Restante Após a Decolagem: {energia_restante} kWh")

Energia Disponível: 1350.0 kWh
Perdas Energéticas (%): 5 %
Perdas Energéticas (kWh): 67.5 kWh
Energia Útil: 1282.5 kWh
Energia Restante Após a Decolagem: 982.5 kWh


### Resultado da análise energética

Considerando as premissas adotadas, a carga atual de 90% corresponde a 1350 kWh de energia disponível. Com perdas energéticas estimadas em 5%, equivalentes a 67,5 kWh, a energia útil disponível é de 1282,5 kWh.

Após o consumo elétrico estimado de 300 kWh durante a fase de decolagem, a espaçonave permaneceria com 982,5 kWh de energia disponível para as etapas seguintes da missão.

A autonomia é analisada nesta simulação em termos de disponibilidade energética. Como não foram fornecidos dados de consumo por unidade de tempo, não é possível estimar a autonomia da espaçonave em horas.

## 7. Considerações finais

A simulação permitiu aplicar os critérios de segurança definidos aos dados de telemetria da Aurora Siger. No cenário analisado, todos os parâmetros permaneceram dentro das condições estabelecidas, resultando na classificação **"PRONTO PARA DECOLAR"**.

A análise energética indicou uma energia disponível inicial de 1350 kWh e, após considerar as perdas energéticas e o consumo elétrico estimado durante a fase de decolagem, uma energia restante de 982,5 kWh.

Os resultados apresentados são baseados nas premissas adotadas para esta simulação e não representam especificações técnicas reais da espaçonave.